framework/infrastructure/traffic_director.py

```diff
@@ -612,30 +628,54 @@ def create_target_proxy(self):
             target_proxy_type,
             self.url_map.name,
         )
-        self.target_proxy = create_proxy_fn(name, self.url_map)
+        
+        if ipv6:
+            self.target_proxy_v6 = create_proxy_fn(name, self.url_map)
+        else:
+            self.target_proxy = create_proxy_fn(name, self.url_map)
 
-    def delete_target_grpc_proxy(self, force=False):
+    def delete_target_grpc_proxy(self, force=False, ipv6=False):
         if force:
-            name = self.make_resource_name(self.TARGET_PROXY_NAME)
+            if ipv6:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME)
+            else:
+                name = self.make_resource_name(self.TARGET_PROXY_NAME + "-v6")
         elif self.target_proxy:
             name = self.target_proxy.name
+        elif self.target_proxy_v6:
+             name = self.target_proxy_v6.name
         else:
             return
```

Comment:
This should be wrapped in `if ipv6:`.  Otherwise, you are relying on the ipv4 being cleaned up before calling the ipv6 cleanup which add fragility.

Reply:
Done